nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -226,6 +226,38 @@ in
             ExecStart = "@${dhcpcd}/sbin/dhcpcd dhcpcd --quiet ${optionalString cfg.persistent "--persistent"} --config ${dhcpcdConf}";
             ExecReload = "${dhcpcd}/sbin/dhcpcd --rebind";
             Restart = "always";
+            # Proc filesystem
+            ProcSubset = "all";
+            ProtectProc = "invisible";
+            # Access write directories
+            UMask = "0027";
+            # Capabilities
+            CapabilityBoundingSet = [ "CAP_NET_ADMIN" "CAP_NET_BIND_SERVICE" "CAP_NET_RAW" "CAP_SETGID" "CAP_SETUID" "CAP_SYS_CHROOT" ];
+            # Security
+            NoNewPrivileges = true;
+            # Sandboxing
+            ProtectSystem = true;
+            ProtectHome = true;
+            PrivateTmp = true;
+            PrivateDevices = true;
+            PrivateUsers = false;
+            ProtectHostname = true;
+            ProtectClock = true;
+            ProtectKernelTunables = false;
```

Comment:
nit: For the ones you opt out of, a short comment why would be interesting.

Reply:
When `ProtectKernelTunables` is active, there is an error in the `dhcpcd` log:
```
dhcpcd[14944]: ipv6nd_applyra: Read-only file system
```

Reply:
For example, these changes work on my test virtual machine:
```diff
diff --git a/nixos/modules/services/networking/dhcpcd.nix b/nixos/modules/services/networking/dhcpcd.nix
index 4cf86653205..bcd6361ec1b 100644
--- a/nixos/modules/services/networking/dhcpcd.nix
+++ b/nixos/modules/services/networking/dhcpcd.nix
@@ -230,13 +230,14 @@ in
             ProcSubset = "all";
             ProtectProc = "invisible";
             # Access write directories
+            ReadWritePaths = [ "/etc/resolv.conf" "/var/db/dhcpcd" ];
             UMask = "0027";
             # Capabilities
             CapabilityBoundingSet = [ "CAP_NET_ADMIN" "CAP_NET_BIND_SERVICE" "CAP_NET_RAW" "CAP_SETGID" "CAP_SETUID" "CAP_SYS_CHROOT" ];
             # Security
             NoNewPrivileges = true;
             # Sandboxing
-            ProtectSystem = true;
+            ProtectSystem = "strict";
             ProtectHome = true;
             PrivateTmp = true;
             PrivateDevices = true;
```
But these parameters do not work in the `nixosTests.networking.scripted.dhcpDefault` test. I can't figure out why. That's why I deleted these changes.